src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+    m_buttonGridLayout->setGeometry(QRect(250, -5, 250, 100));
+}
+
+void MainWindow::terminateApplication()
+{
+    // Terminate the application
+    QCoreApplication::quit();
+}
+
+void MainWindow::createConfigFileGenerateFailureBox()
+{
+    // Create a message box with appropriate displayed text
+    QMessageBox configFileFailBox;
+    configFileFailBox.setText(m_CONFIG_FILE_GENERATE_FAIL_BOX_PRIMARY_TEXT);
+    configFileFailBox.setInformativeText(m_CONFIG_FILE_GENERATE_FAIL_BOX_INFO_TEXT);
+    configFileFailBox.setStandardButtons(QMessageBox::Ok);
+
+    int configFileFailBoxSelection = configFileFailBox.exec();
+
+    switch(configFileFailBoxSelection)
+    {
+    // If the user presses "Ok", terminate the application
+    case QMessageBox::Button::Ok :
+    {
+        emit conditionToTerminateMet();
+        break;
+    }
+
+    default :
+    {
+        // Terminate the application by default
+        emit conditionToTerminateMet();
+        break;
+    }
+    }
+}
+
+void MainWindow::createInvalidKeyBox()
+{
+    // Create the invalid key message box
+    QMessageBox invalidKeyBox;
+    invalidKeyBox.setText(m_INVALID_KEY_BOX_PRIMARY_TEXT);
+    invalidKeyBox.setInformativeText(m_INVALID_KEY_BOX_INFO_TEXT);
+    invalidKeyBox.setStandardButtons(QMessageBox::Ok);
+
+    int invalidKeyBoxSelection = invalidKeyBox.exec();
+
+    switch(invalidKeyBoxSelection)
+    {
+    // If the user presses the "Ok" button
+    case QMessageBox::Button::Ok :
+    {
+        // Terminate the application
+        emit conditionToTerminateMet();
+        break;
+    }
+
+    default :
+    {
+        // Terminate the application in the default case
+        emit conditionToTerminateMet();
+        break;
+    }
+    }
+}
+
+void MainWindow::createTableWidgetUsingMap()
+{
+    // Set the number of rows in the table widget to be the number of bills stored in the bill map
+    m_billTableWidget->setRowCount(m_billMap.size());
+
+    // Set the number of columns to the number of fields which each Bill displays
+    m_billTableWidget->setColumnCount(4);
+
+    // Set the table widget headers to the appropriate fields each Bill displays
+    m_billTableWidget->setHorizontalHeaderLabels(QString(m_BILL_NAME_COLUMN_HEADER_STRING + ";" + m_BILL_AMOUNT_DUE_COLUMN_HEADER_STRING + ";" + m_BILL_DUE_DATE_COLUMN_HEADER_STRING + ";" + m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING).split(";"));
+
+    // Initialize the row we're setting to zero
+    int row = 0;
+
+    // Create an iterator to iterate over the bill map
+    QMap<QString, Bill>::iterator billMapIterator;
+
+    for(billMapIterator = m_billMap.begin(); billMapIterator != m_billMap.end(); ++billMapIterator)
+    {
+        // Retrieve the current Bill being checked
+        Bill currentBill = *billMapIterator;
+
+        // Create a date edit for displaying and allowing the editing of the bill's due date
+        QDateEdit *dateEdit = new QDateEdit(this);
```

Comment:
Do `dateEdit` and `fundedStatusBox` need to be deleted as they're dynamically allocated pointers, despite being created in a function and not the constructor?

Reply:
They should be deleted as they are created dynamically. However they're needed for display in the QTableWidget, attempting to delete them in and outside of the for loop in this function results in a segmentation fault.

However, the deletion can occur by calling the `clear()` function of QTableWidget in the destructor.